Kernel/Devices/TTY/MasterPTY.cpp

```diff
@@ -49,24 +51,33 @@ MasterPTY::~MasterPTY()
     PTYMultiplexer::the().notify_master_destroyed({}, m_index);
 }
 
+bool MasterPTY::has_slave() const
+{
+    return m_slave.with([](auto& slave) -> bool {
+        return slave;
+    });
+}
```

Comment:
Is this correct?

As soon as `has_slave()` exits, we're no longer in the lock and `m_slave` might change before the caller goes to use it.